docs/design/HealthEndpoint.md

```diff
@@ -0,0 +1,311 @@
+# Design Document: DAB Health Endpoint
+## Objective:
+The objective of this task is to enhance the "Health Endpoint" for DAB. Currently we only show whether the DAB Engine is healthy or not with a small description of the version.\
+However, the objective under this task item is to enhance the health endpoint to support detailed description of each enabled entity by performing health checks on them based on different kinds of parameters that would be defined below.
+
+## Need for DAB Health Check Endpoint
+Azure App Service & Azure Kubernetes Service (AKS) support health probes to monitor the health of your application. If a service fails health checks, Azure can automatically restart it or redirect traffic to healthy instances.\
+Similarly, we need Health Endpoint for Data API builder because if it fails health checks in a way a customer deems past a threshold, they have the option to recycle the container or send an alert to direct engineers.
+
+## Current Setup
+There is no official industry standard for the health endpoint. /health or variations like /_health are common by convention.\
```

Comment:
Could you more clearly state that this feature will have two endpoints? / and /health. That / (the root) returns what it already returns today, the basic health) and that /health returns the more comprehensive health depending on a mixture of if health is enabled and the user meets the roles requirement?